### PR TITLE
feat(value-list-item, pick-list-item): adds start slots, uses conventional slot names, removes non-conventional slot names

### PR DESCRIPTION
**Pull request by asangma**
DRAFT : hold until #903 is murgled. Will rebase to `master`.

**Related Issue:** #1039

## Summary
- Existing slot `secondary-action` still works
- Adds slot with conventionally named slot `actions-end` to replace `secondary-action`
- Adds `actions-start` slot

This addresses regularly requested UI requirements.

TODO: 

- [x] documentation on deprecated slot `secondary-action`
- [x] review :hover state UI with design team

### CODE REVIEW

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx [hunk 1]
```diff
@@ -243,26 +244,39 @@ export class CalcitePickListItem {
...
 
-  renderSecondaryAction(): VNode {
-    const hasSecondaryAction = getSlotted(this.el, SLOTS.secondaryAction);
-    return hasSecondaryAction || this.removable ? (
-      <div class={CSS.action}>
-        <slot name={SLOTS.secondaryAction}>{this.renderRemoveAction()}</slot>
+  renderActionsStart(): VNode {
+    const { el } = this;
+    const hasActionsStart = getSlotted(el, SLOTS.actionsStart);
+    console.log(hasActionsStart);
```
Comment (driskull) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:261]: remove zeee console log

File: src/components/calcite-value-list-item/calcite-value-list-item.tsx [hunk 2]
```diff
@@ -2,11 +2,15 @@ import { Component, Element, Host, Method, Prop, h, VNode, Listen } from "@stenc
...
 import { CSS } from "../calcite-pick-list-item/resources";
-import { ICONS } from "./resources";
+import { ICONS, SLOTS } from "./resources";
+import { SLOTS as PICK_LIST_SLOTS } from "../calcite-pick-list-item/resources";
+import { getSlotted } from "../../utils/dom";
 
 /**
- * @slot secondary-action - A slot intended for adding a `calcite-action` or `calcite-button`. This is placed at the end of the item.
- */
+  * @slot secondary-action - **[DEPRECATED]** use actions-end instead.
```
Comment (driskull) [inline src/components/calcite-value-list-item/calcite-value-list-item.tsx:10]: Do we need to deprecate if this isnt really a change?

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx (hunk 1 above)
Comment (driskull) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:261]: remove zee console log

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx [hunk 3]
```diff
@@ -15,9 +15,10 @@ import { ICON_TYPES } from "../calcite-pick-list/resources";
 import { getSlotted } from "../../utils/dom";
 
 /**
- * @slot secondary-action - A slot intended for adding a `calcite-action` or `calcite-button` to the right side of the card.
- * This is placed at the end of the item.
- */
+  * @slot secondary-action - **[DEPRECATED]** use actions-end instead.
```
Comment (driskull) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:18]: Is this really a deprecation?

**Review by driskull — CHANGES_REQUESTED** [review]
Looking good

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:261]: LOL. 🧹

Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:261]: 🧹

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx (hunk 3 above)
Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:18]: It would be nice if it were to go away in the next release or two. :)

**Comment by asangma** [comment]
@driskull I unintentionally resolved a conversation. 😬

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx (hunk 3 above)
Comment (driskull) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:18]: Can we just remove them now and note them in the migration steps?

Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx:18]: I'll do it!

**Comment by asangma** [comment]
@driskull @jcfranco Updated with the `secondary-action` slot removed.

A heap of demos need to be refactored to accommodate for the removal of the `text-` prefix on string properties. I would like to do that in a different PR. There's a heap-o-heap of those demos.

**Review by driskull — APPROVED** [review]
👍

**Comment by asangma** [comment]
@jcfranco If you have no objections, I shall murgle.

**Comment by jcfranco** [comment]
@asangma Sorry for the delay. I'll submit my review by the end of the day. #unintentionalrhyme

**Review by jcfranco — APPROVED** [review]
👍 

Small note, the commit message may need a little change. The non-conventional slots have been removed vs deprecated.

**Comment by asangma** [comment]
Thanks @driskull and @jcfranco.
I'll adjust the commit message and murgle!

**Pull request merged**